packages/embeds/embed-core/src/EmbedElement.ts

```diff
@@ -197,20 +196,26 @@ export class EmbedElement extends HTMLElement {
     skeletonEl.innerHTML = skeletonContent;
   }
 
+  public setTheme(theme: EmbedThemeConfig | null) {
+    const allPossibleThemeClasses = [EMBED_DARK_THEME_CLASS, EMBED_LIGHT_THEME_CLASS];
+
+    const newThemeClass = getThemeClassForEmbed({ theme });
+
+    if (newThemeClass === this.themeClass) {
+      return;
+    }
+    this.themeClass = newThemeClass;
+    this.classList.remove(...allPossibleThemeClasses);
+    this.classList.add(this.themeClass);
+  }
```

Comment:
Centralize theme setting logic. It fixes an existing bug where the class wasn't changing on system theme change.